projects/media-viewer/src/lib/toolbar/left-pane/left-pane.component.ts

```diff
@@ -18,10 +19,41 @@ export class ToolbarLeftPaneComponent {
   @Input() showNavigationBtns: boolean;
   @Input() sidebarOpen: BehaviorSubject<boolean>;
   @Input() searchBarHidden: BehaviorSubject<boolean>;
-  @Input() drawMode: BehaviorSubject<boolean>;
-  @Input() highlightMode: BehaviorSubject<boolean>;
+  // Local presentation state that can be pre-initialised from the template
+  @Input() highlightMode = false;
+  @Input() drawMode = false;
+
+  // local array of any subscriptions so that we can tidy them up later
+  private subscriptions: Subscription[] = [];
+  constructor(private readonly toolbarEventsService: ToolbarEventsService) {}
+
+  ngOnInit(): void {
+    // Listen for any changes invoked on the toolbar events Service and initialise any default behaviour state
+    this.subscriptions.push(this.toolbarEventsService.highlightMode.subscribe((toggleValue) => {
```

Comment:
could we use the async pipe here, as it's only for view state

Reply:
i’ve conceded in this example… its now an async pipe. @benouaer can you recheck?